astropy/cosmology/core.py

```diff
@@ -2562,6 +2867,13 @@ def _float_or_none(x, digits=3):
     fmtstr = "{0:.{digits}g}".format(x, digits=digits)
     return fmtstr.format(x)
 
+def vectorize_if_needed(func, x):
+    """ Helper function to vectorize functions on array inputs"""
+    if not isiterable(x):
+        return func(x)
+    else:
+        return np.vectorize(func)(x)
+
```

Comment:
Ok. I also tested this and the speedup is not so important, so it's fine as you have it.